bot/update_tracker.py

```
@@ -1,146 +1,613 @@
+            temp_file = f"{self.tracker_file}.tmp"
+            try:
+                with open(temp_file, "w", encoding='utf-8') as f:
+                    json.dump(data, f)
+                os.replace(temp_file, self.tracker_file)
+                    
+            except OSError as e:
+                if os.path.exists(temp_file):
+                    try:
+                        os.remove(temp_file)
+                    except OSError:
+                        pass
```

Ruff

Use `contextlib.suppress(OSError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(OSError)`

(SIM105)